src/saltext/vmware/modules/esxi.py

```diff
@@ -1166,6 +1166,210 @@ def get_user(
         raise salt.exceptions.SaltException(str(exc))
 
 
+def add_role(
+    role_name,
+    privilege_ids,
+    esxi_host_name=None,
+    service_instance=None,
+):
+    """
+    Add local role to service instance, which may be an ESXi host or vCenter instance.
+
+    role_name
+        Role to create on ESXi host. (required).
+
+    privilege_ids
+        List of privileges for the role. (required).
+        Refer: https://docs.vmware.com/en/VMware-vSphere/7.0/com.vmware.vsphere.security.doc/GUID-ED56F3C4-77D0-49E3-88B6-B99B8B437B62.html
+        Example: ['Folder.Create', 'Folder.Delete'].
+
+    esxi_host_name
```

Comment:
"Connect to this ESXi host using your pillar's service_instance credentials."?

Reply:
ok will change